src/Compilers/Core/Portable/Symbols/INamedTypeSymbol.cs

```diff
@@ -134,5 +134,8 @@ public interface INamedTypeSymbol : ITypeSymbol
         /// If false, the symbol does not contain extension methods. 
         /// </summary>
         bool MightContainExtensionMethods { get; }
+
+        ImmutableArray<ITypeSymbol> TupleElementTypes { get; }
+        ImmutableArray<string> TupleElementNames { get; }
```

Comment:
What happens when I call these methods on non-tuple types?  For instance what happens when this is invoked on say `System.String`?  

Do we have tests for that case?